assets/js/util/test/numberFormat.js

```diff
@@ -115,5 +116,98 @@ describe( 'numberFormat', () => {
 	it.each( siteKitLocales )( 'formats numbers correctly with locale variant %s', ( locale, value, expected ) => {
 		setupGoogleSiteKit( locale );
 		expect( numberFormat( value ) ).toStrictEqual( expected );
+		expect( console ).not.toHaveWarned();
+	} );
+
+	const NumberFormat = Intl.NumberFormat;
+
+	/*
+	 * This function mocks the implementation of certain errors identified in
+	 * https://github.com/google/site-kit-wp/issues/3255
+	 * so that we can test that we've properly handled them.
+	 */
+	const mockNumberFormat = () => {
+		global.Intl.NumberFormat = ( locales, options ) => {
+			if ( undefined === options ) {
+				return NumberFormat( locales, options );
+			}
+			const optionsThatError = {
+				currencyDisplay: 'narrow',
+				currencySign: 'accounting',
+				style: 'unit',
+				signDisplay: '*', // * denotes any value.
+				compactDisplay: '*',
+			};
+			for ( const [ key, value ] of Object.entries( options ) ) {
+				if ( optionsThatError[ key ] && [ value, '*' ].includes( optionsThatError[ key ] ) ) {
+					throw new Error( 'Implementation error' );
+				}
+			}
```

Comment:
Let's export these things from the module so we're not maintaining these in multiple places.

How about we extract all of this to a single function like `omitUnstableNumberFormatOptions( formatOptions ) : numberFormatOptionsWithoutTheUnstableOnes` (a la [`_.omit`](https://lodash.com/docs/4.17.15#omit))? This should be annotated as `@private`.

Reply:
That won't make sense in this scenario as our test isn't intended to clean up the formatOptions, that's the role of `numberFormat`. Instead we're trying to imitate the browser behaviour and throw the appropriate error as Safari would in these instances so that it triggers the catch clause in our `numberFormat` function.

Reply:
Yeah, my mistake; this loop throws rather than modifying the options 😄  Originally I was looking at just extracting the object(s) of unstable options (since these are constant) and saw the loop was duplicated from the original implementation. See my review for my final thoughts here.